Tools/TestWebKitAPI/Tests/mac/ContextMenuTests.mm

```diff
@@ -150,6 +150,27 @@ - (void)rightClick:(NSPoint)clickLocation andSelectItemMatching:(MenuItemFilter)
     TestWebKitAPI::Util::run(&selectedItem);
 }
 
+- (_WKContextMenuElementInfo *)rightClickAtPointAndWaitForContextMenu:(NSPoint)clickLocation
+{
+    auto uiDelegate = adoptNS([TestUIDelegate new]);
+
+    __block RetainPtr<_WKContextMenuElementInfo> result;
+    __block bool gotProposedMenu = false;
+    [uiDelegate setGetContextMenuFromProposedMenu:^(NSMenu *, _WKContextMenuElementInfo *elementInfo, id<NSSecureCoding>, void (^completion)(NSMenu *)) {
+        result = elementInfo;
+        gotProposedMenu = true;
+        completion(nil);
+    }];
+
```

Comment:
Nit - it _might_ be worth a `RELEASE_ASSERT(!self.UIDelegate);` before getting here, to make it clear that this helper method overwrites any UIDelegate that's previously set.

Reply:
Done!